impeller/entity/contents/content_context.cc

```diff
@@ -548,4 +547,25 @@ void ContentContext::ClearCachedRuntimeEffectPipeline(
   }
 }
 
+void ContentContext::RecordCommandBuffer(
+    std::shared_ptr<CommandBuffer> command_buffer) const {
+  pending_command_buffers_->command_buffers.push_back(
+      std::move(command_buffer));
+  // Metal systems seem to have a limit on the number of command buffers that
```

Comment:
This is the big change.

Reply:
This still seems arbitrary. Can we flush pending command buffers when we can't create more or is there a delay between the flush and when we can acquire more buffers.

Reply:
Or have the magic number of 10 as well as the flush and retry on allocation failure. Just having a backend specific magic number seems icky.

Reply:
Thinking about it more I should just make it immediately flush on metal, since we get no benefit from batching.

Reply:
unfortunately there is no allocation failure, instead `[queue commandBufferWithDescriptor:desc];` appears to hang.

Reply:
removed the limit, now we Record -> Flush.

Reply:
Oh well. Not having the arbitrary limit looks cleaner though.

Reply:
Yeah but it doesn't really matter for metal, I don't think we'll have a way to batch submit with that API ever.